### PR TITLE
feat: methods for eager endpoints

### PR DESCRIPTION
**Pull request by BeeMargarida**
| - | - |
| --- | --- |
| Issue | Related to https://github.com/ripe-tech/ripe-twitch/pull/94 |
| Dependencies | https://github.com/ripe-tech/ripe-twitch/pull/94 |
| Decisions | `getEagerCollection`, `getEagerDrop` |
| Animated GIF | -- |

### CODE REVIEW

**Comment by gcandal** [comment]
Types are missing

File: types/drop.d.ts [hunk 1]
```diff
@@ -20,6 +20,18 @@ export interface Drop {
     readonly meta: Record<string, unknown>;
 }
 
+export interface DropEager {
```
Comment (gcandal) [inline types/drop.d.ts]: Can't you write this in terms of `Drop` and override just the `products` field?

File: types/collection.d.ts [hunk 2]
```diff
@@ -41,6 +44,7 @@ export declare class CollectionAPI {
     listCollections(options: APIOptions): Promise<Collection[]>;
     createCollection(payload: CollectionCreate): Promise<Collection>;
     getCollection(id: string): Promise<Collection>;
+    getEagerCollection(id: string): Promise<Omit<Collection, "products" | "streamers"> & { streamers: Streamer[], products: ProductCollection[] }>;
```
Comment (gcandal) [inline types/collection.d.ts]: Didn't mean to mislead you, but what I meant was doing what you just did (using `Omit`) but keeping the interface you had (`EagerCollection`?)

Comment (BeeMargarida) [reply types/collection.d.ts]: Oh, got it!

**Review by gcandal — APPROVED** [review]

**Review by joamag — APPROVED** [review]

**Pull request merged**